api/fbs/v1/write.fbs

```diff
@@ -0,0 +1,60 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+include "common.fbs";
+include "database.fbs";
+
+namespace banyandb.v1;
+
+table String {
+    value: string;
+}
+
+table Int {
+    value: int64;
+}
+
+union FieldValue {
+    String,
+    Int
+}
+
+table Field {
+    value: FieldValue;
+}
+
+table entityValue {
+    // entity_id could be span_id of a Span or segment_id of a Segment in the context of Trace
+    entity_id: string;
+    // timestamp_nanoseconds is in the timeunit of nanoseconds. It represents
+    // 1) either the start time of a Span/Segment,
+    // 2) or the timestamp of a log
+    timestamp_nanoseconds: uint64;
+    // binary representation of segments, including tags, spans...
+    data_binary: [ubyte];
+    // support all of indexed fields in the fields.
+    fields: [Field];
```

Comment:
Could you elaborate on why you only write values instead of key-value paires? Pls, comment on `fileds` to show more details.

Reply:
Consider using `Pair`?

Reply:
@hanahmily Should we consider setting field name once in whole streaming?

Reply:
The keys of fields(indexed ones) are defined in the series's metadata component. We don't have to do metadata exchange here.

The old metadata exchange path is based on tags that are removed from the query execution plan, being merged into the `data_binary`

Reply:
> The keys of fields(indexed ones) are defined in the series's metadata component. We don't have to do metadata exchange here.

But how do you know the field name as this is basically an array only. Such as which element is `traceid`, or `endpoint name`.

Reply:
For example.
The metadata is `[traceid, endpoint_name]`, the index **0** is `traceid`, and index **1** is `endpoint_name`. Once [`xxx-aa`, `/query`] gets written, based on the order of fields in metadata, banyandb will know `xxx-aa` is the `traceid`, because their index is the No.0.

Reply:
This concept works, but I can't see this in `Metadata`.

Reply:
The metadata I mentioned is actually the combination of the IndexRule and IndexRuleBinding. I will avoid to use "metadata" to mention it.

Reply:
OK, you mean this is defined in the schema? Back to creating the table?

Reply:
Exactly. Based on them, we don't define the key in write protocol

Reply:
> Consider using `Pair`?

If we use `Pair` instead `Field`, it seems to conflict with `Pair` in the `query`.

Reply:
> > Consider using `Pair`?
> 
> If we use `Pair` instead `Field`, it seems to conflict with `Pair` in the `query`.

2 things. One we don't need `Pair` here according to Gao's response, we have the metadata in place. Two, I used to mean using the `Pair` if the field name is required